hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/ListingBasedRollbackStrategy.java

```diff
@@ -239,7 +239,15 @@ private FileStatus[] fetchFilesFromCommitMetadata(HoodieInstant instantToRollbac
     SerializablePathFilter pathFilter = getSerializablePathFilter(baseFileExtension, instantToRollback.getTimestamp());
     Path[] filePaths = getFilesFromCommitMetadata(basePath, commitMetadata, partitionPath);
 
-    return fs.listStatus(filePaths, pathFilter);
+    return fs.listStatus(Arrays.stream(filePaths).filter(entry -> {
+      try {
+        return fs.exists(entry);
+      } catch (IOException e) {
+        LOG.error("Exists check failed for " + entry.toString(), e);
+      }
+      // if IOException is thrown, do not ignore. lets try to add the file of interest to be deleted. we can't miss any files to be rolled back.
```

Comment:
@nsivabalan does n't the line above just ignore the IOException? Why would fs.listStatus() return stuff that does not exist?

Reply:
This is a misleading comment. This PR intends to ignore files that have already been deleted by the cleaner and gracefully handle the FileNotFound/IO exception.
@XuQianJin-Stars Since you reviewed this PR, can you validate my understanding?